file/validate.go

```diff
@@ -1,20 +1,70 @@
 package file
 
 import (
+	"encoding/json"
 	"fmt"
 
 	ghodss "github.com/ghodss/yaml"
 	"github.com/kong/deck/utils"
 	"github.com/xeipuuv/gojsonschema"
 )
 
+func check(item string, t *string) bool {
+	if t != nil || len(*t) > 0 {
+		fmt.Println(item + "[err] is not allowed to be specified.")
```

Comment:
I think we'd want to instead aggregate the unacceptable fields `checkDefaults()` sees and return an error if we see more than 0. Something like:

```
func checkDefaults(kd KongDefaults) error {
    var invalid string[]
    if kd.Service.ID != nil || len(*kd.Service.ID) > 0 {
        invalid = append(invalid, "Service.ID")
    }
    ...
    if len(invalid) > 0 {
        return fmt.Errorf("unacceptable fields in defaults: %s", strings.Join(invalid, ", "))
    }
    return nil
}
```

Reply:
Yeah the example was written directly into the comment without the benefit of a syntax checker :)

Is the gist of the change clear though?